modules/web-console/frontend/app/components/page-queries/components/queries-notebook/controller.ts

```diff
@@ -1904,6 +2057,22 @@ export class NotebookCtrl {
                 $modal({scope, templateUrl: messageTemplateUrl, show: true});
             }
         };
+
+        $window.addEventListener('beforeunload', () => {
```

Comment:
1. Every subsequent page enter will add new event listener. You should at least remove it when leaving the state.
2. I'm not sure this should be a thing at all. Maybe consider cancelling queries when client session is over (but this might cause issues with faulty network connection). Or just leave queries running, we don't cancel any other requests or long running operations (or do we?).

Reply:
This listener will most likely leak references.